### PR TITLE
Feat/client

### PR DESCRIPTION
**Pull request by efesurekli**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,51 +22,100 @@ Skyrim allows you to run any large weather model with a consumer grade GPU. Unti
...
-1. Start with a docker base image `pytorch/pytorch:2.2.2-cuda11.8-cudnn8-devel`.
-2. Then clone the repo and run `./build.sh`.
+### Forecasting using Modal:
 
-### vast.ai setup
+If you are running on modal then run:
+`modal run modal/forecast.py:run_inference`
+
+This by default uses `pangu` model to forecast for the next 6 hours, starting from yesterday. It gets initial conditions from ECMWF IFS and writes the forecast to a modal volume. You can explore the forecast by running a notebook (without GPU) in modal:
+
```
Comment (m13uz) [inline README.md:37]: We need to debug the ifs initial conditions 🥲

File: README.md [hunk 2]
```diff
@@ -23,51 +22,100 @@ Skyrim allows you to run any large weather model with a consumer grade GPU. Unti
...
+## Large weather models supported
 
 Currently supported models are:
 
 - [x] [Graphcast](https://arxiv.org/abs/2212.12794)
 - [x] [Pangu](https://arxiv.org/abs/2211.02556)
 - [x] [Fourcastnet](https://arxiv.org/abs/2202.11214) (v1 & v2)
 - [x] [DLWP](https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2020MS002109)
-- [ ] [Fuxi](https://www.nature.com/articles/s41612-023-00512-1)
+- [x] [Fuxi](https://www.nature.com/articles/s41612-023-00512-1)
```
Comment (m13uz) [inline README.md:118]: for now
```suggestion
- [ ] [Fuxi](https://www.nature.com/articles/s41612-023-00512-1)
```

File: skyrim/core/models/base.py [hunk 3]
```diff
@@ -71,60 +68,32 @@ def predict_one_step(
...
-                    pred_time=pred_time,
-                    ic_source=ic_source,
-                    output_dir=output_dir,
+                output_path = save_forecast(
+                    pred,
+                    self.model_name,
+                    start_time,
+                    pred_time,
+                    source,
+                    config=save_config,
```
Comment (m13uz) [inline skyrim/core/models/base.py:91]: ❤️

File: skyrim/core/skyrim.py [hunk 4]
```diff
@@ -57,8 +57,8 @@ def predict(
         date: str,  # YYYMMDD, e.g. 20180101
         time: str,  # HHMM, e.g. 0300, 1400, etc
         lead_time: int = 6,  # in hours 0-24, will be clipped to nearest 6 multiple
-        save: bool = True,
-        output_dir: str | Path = OUTPUT_DIR,
+        save: bool = False,
+        save_config: dict = {},
```
Comment (m13uz) [inline skyrim/core/skyrim.py:61]: some docstring about `save_config` might be useful

File: skyrim/common.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        path += "/"
+    s3_client = boto3.client("s3")
+    p = urlparse(path)
+    return "Contents" in s3_client.list_objects_v2(
+        Bucket=p.netloc, Prefix=p.path[1:], MaxKeys=1
+    )
+
+
+def save_forecast(
+    pred: xr.DataArray | xr.Dataset,
```
Comment (m13uz) [inline skyrim/common.py:75]: Currently, all model predictions are returned as `xr.DataArray`  types except the graphcast afair. We better homogenize the data types in the near future ⏳

File: skyrim/core/models/pangu.py [hunk 6]
```diff
@@ -34,7 +34,7 @@ class PanguModel(GlobalModel):
 
     Parameters:
     - model_name (str): Name of the model, default is 'pangu'.
-    """
+"""
```
Comment (m13uz) [inline skyrim/core/models/pangu.py:37]:
```suggestion
    """
```

**Review by m13uz — APPROVED** [review]
LFG!

File: skyrim/common.py (hunk 5 above)
Comment (efesurekli) [reply skyrim/common.py:75]: agree, also this can be used within Prediction so that we do prediction.save(...)

**Pull request merged**